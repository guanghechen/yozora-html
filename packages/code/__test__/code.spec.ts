import type { Code } from '@yozora/ast'
import renderCode from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const code: Code = {
      type: 'code',
      lang: '',
      meta: '',
      value: 'multiple\nliteral\ntexts',
    }
    expect(renderCode(code)).toMatchSnapshot()
  })

  it('javascript', function () {
    const code: Code = {
      type: 'code',
      lang: 'javascript',
      meta: '',
      value: 'const a = 1\nconst b = 2\nconsole.log("a + b =", a + b)\n',
    }
    expect(renderCode(code)).toMatchSnapshot()
  })
})
