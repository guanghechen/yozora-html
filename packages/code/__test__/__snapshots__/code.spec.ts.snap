// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `
"<code class="yozora-code"><pre>multiple
literal
texts</pre></code>"
`;

exports[`snapshot javascript 1`] = `
"<code class="yozora-code"><pre><span class="token keyword">const</span> a <span class="token operator">=</span> <span class="token number">1</span>
<span class="token keyword">const</span> b <span class="token operator">=</span> <span class="token number">2</span>
console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">"a + b ="</span><span class="token punctuation">,</span> a <span class="token operator">+</span> b<span class="token punctuation">)</span>
</pre></code>"
`;
